structuredheadings/plugin.js

```diff
@@ -34,7 +34,7 @@
     var style = editor.config.autonumberStyles[styleName];
     if (element.type === CKEDITOR.NODE_ELEMENT) {
       clearStyles(editor, element);
-      if (style) {
+      if (style && style[element.getName()]) {
```

Comment:
alternately, 

`elementName = element.getName()` above, and then use elementName rather than getName() twice.  though I think actual perf implications would be minimal